contracts/core/tokens/rebalancing-libraries/StandardStartRebalanceLibrary.sol

```diff
@@ -0,0 +1,313 @@
+/*
+    Copyright 2018 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.4.25;
+pragma experimental "ABIEncoderV2";
+
+import { Math } from "openzeppelin-solidity/contracts/math/Math.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { AddressArrayUtils } from "../../../lib/AddressArrayUtils.sol";
+import { IAuctionPriceCurve } from "../../lib/auction-price-libraries/IAuctionPriceCurve.sol";
+import { ICore } from "../../interfaces/ICore.sol";
+import { ISetToken } from "../../interfaces/ISetToken.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+import { RebalancingHelperLibrary } from "../../lib/RebalancingHelperLibrary.sol";
+
+/**
+ * @title StandardProposeLibrary
+ * @author Set Protocol
+ *
+ * Default implementation of Rebalancing Set Token startRebalance function
+ */
+
+
+library StandardStartRebalanceLibrary {
+    using SafeMath for uint256;
+    using AddressArrayUtils for address[];
+
+    /* ============ Structs ============ */
+    struct StartRebalanceParameters {
+        address currentSet;
+        address nextSet;
+        address auctionLibrary;
+        uint256 proposalStartTime;
+        uint256 proposalPeriod;
+        ICore coreInstance;
+        IVault vaultInstance;
+        RebalancingHelperLibrary.State rebalanceState;
+    }
+
+    struct BiddingParameters {
+        uint256 minimumBid;
+        uint256 remainingCurrentSets;
+        uint256[] combinedCurrentUnits;
+        uint256[] combinedNextSetUnits;
+        address[] combinedTokenArray;
+    }
+
+    struct SetsDetails {
+        uint256 currentSetNaturalUnit;
+        uint256 nextSetNaturalUnit;
+        uint256[] currentSetUnits;
+        uint256[] nextSetUnits;
+        address[] currentSetComponents;
+        address[] nextSetComponents;
+    }
+
+    /* ============ Internal Functions ============ */
+
+    /**
+     * Function used to validate inputs to propose function and initialize biddingParameters struct
+     *
+     * @param _startRebalanceParameters            Rebalancing Set Token state parameters needed to execute logic
+     * @return                                     Struct containing bidding parameters
+     */
+    function startRebalance(
+        StartRebalanceParameters memory _startRebalanceParameters
```

Comment:
Are you running into stack errors here? Otherwise, we shouldn't have a struct that is only used between RebalancingToken#startRebalance and here